src/main/java/com/ibm/watson/modelmesh/Metrics.java

```diff
@@ -342,13 +368,21 @@ public void logTimingMetricSince(Metric metric, long prevTime, boolean isNano) {
         }
 
         @Override
-        public void logTimingMetricDuration(Metric metric, long elapsed, boolean isNano) {
-            ((Histogram) metricsMap.get(metric)).observe(isNano ? elapsed / M : elapsed);
+        public void logTimingMetricDuration(Metric metric, long elapsed, boolean isNano, String modelId) {
+            if (enablePerModelMetrics) {
```

Comment:
We could accept a `null` `modelId` here so that it can be omitted for metrics where it isn't applicable. The check would then be `if (enablePerModelMetrics && modelId != null)`

Reply:
fixed